builtin/logical/pki/cert_util.go

```diff
@@ -882,6 +884,78 @@ func generateCreationBundle(b *backend, data *dataBundle) error {
 		}
 	}
 
+	URIs := []*url.URL{}
+	var uriAltInt interface{}
+	{
+		if data.csr != nil && data.role.UseCSRSANs {
+			if len(data.csr.URIs) > 0 {
+				if len(data.role.AllowedURISANs) == 0 {
+					return errutil.UserError{Err: fmt.Sprintf(
+						"URI Subject Alternative Names are not allowed in this role, but were provided via CSR"),
+					}
+				}
+
+				// validate uri sans
+				valid := false
+				for _, uri := range data.csr.URIs {
+					for _, allowed := range data.role.AllowedURISANs {
+						validURI := glob.Glob(allowed, uri.String())
+						if validURI {
+							valid = true
+							break
+						}
+					}
+				}
+
+				if !valid {
+					return errutil.UserError{Err: fmt.Sprintf(
+						"URI Subject Alternative Names were provided via CSR which are not valid for this role"),
+					}
+				}
+
+				URIs = data.csr.URIs
+			}
+		} else {
+			uriAltInt, ok = data.apiData.GetOk("uri_sans")
+			if ok {
+				uriAlt := uriAltInt.(string)
+				if len(uriAlt) != 0 {
+					if len(data.role.AllowedURISANs) == 0 {
+						return errutil.UserError{Err: fmt.Sprintf(
+							"URI Subject Alternative Names are not allowed in this role, but were provided via CSR"),
+						}
+					}
+
+					for _, v := range strings.Split(uriAlt, ",") {
```

Comment:
Rather than do this, please use TypeCommaStringSlice for the parameter which will return you a []string already.